src/features/modals/Modal.tsx

```diff
@@ -0,0 +1,48 @@
+import { ReactNode } from "preact/compat";
+
+type ModalProps = {
+  children: ReactNode;
```

Comment:
I prefer using ReactNode over JSX.Element. I used to be able to tell you why, but I remember it was a pretty good reason.